internal/resources/diffValidation/resource_nsxt_lb_profile.go

```diff
@@ -0,0 +1,173 @@
+		diff: diff,
+	}
+}
+
+func (l *LoadBalancerProfile) DiffValidate() error {
+	serviceType := l.diff.Get(serviceTypes)
+	switch serviceType {
+	case LBHttpProfile:
+		err := l.validateProfile(httpProfile, LBHttpProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validateApplication(httpProfile, LBHttpProfile)
+		if err != nil {
+			return err
+		}
+
+	case LBFastTcpProfile:
+		err := l.validateProfile(tcpProfile, LBFastTcpProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validateApplication(tcpProfile, LBFastTcpProfile)
+		if err != nil {
+			return err
+		}
+	case LBFastUdpProfile:
+		err := l.validateProfile(udpProfile, LBFastUdpProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validateApplication(udpProfile, LBFastUdpProfile)
+		if err != nil {
+			return err
+		}
+	case LBCookiePersistenceProfile:
+		err := l.validateProfile(cookieProfile, LBCookiePersistenceProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validatePersistence(cookieProfile, LBCookiePersistenceProfile)
+		if err != nil {
+			return err
+		}
+	case LBSourceIpPersistenceProfile:
+		err := l.validateProfile(sourceIpProfile, LBSourceIpPersistenceProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validatePersistence(sourceIpProfile, LBSourceIpPersistenceProfile)
+		if err != nil {
+			return err
+		}
+	case LBGenericPersistenceProfile:
+		err := l.validateProfile(genericProfile, LBGenericPersistenceProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validatePersistence(genericProfile, LBGenericPersistenceProfile)
+		if err != nil {
+			return err
+		}
+	case LBClientSslProfile:
+		err := l.validateProfile(clientProfile, LBClientSslProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validateSsl(clientProfile, LBClientSslProfile)
+		if err != nil {
+			return err
+		}
+	case LBServerSslProfile:
+		err := l.validateProfile(serverProfile, LBServerSslProfile)
+		if err != nil {
+			return err
+		}
+
+		err = l.validateSsl(serverProfile, LBServerSslProfile)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (l *LoadBalancerProfile) validateApplication(profileTypes string, service_type string) error {
+	p := l.diff.Get(profileTypes)
+	for _, profile := range p.([]interface{}) {
+		profile_type := profile.(map[string]interface{})["profile_type"].(string)
+		if profile_type != applicationProfile {
+			return fmt.Errorf("please provide profile_type as " + applicationProfile + " " + "for serviceType" + " " + service_type)
+		}
+	}
+	return nil
+}
+
+func (l *LoadBalancerProfile) validatePersistence(profileTypes string, service_type string) error {
+	p := l.diff.Get(profileTypes)
+	for _, profile := range p.([]interface{}) {
+		profile_type := profile.(map[string]interface{})["profile_type"].(string)
+		if profile_type != persistenceProfile {
+			return fmt.Errorf("please provide profile_type as " + persistenceProfile + " " + " for serviceType" + " " + service_type)
+		}
+	}
+	return nil
```

Comment:
what is difference between these functions ? is it anyway different that the lb monitor scenario? why different functions here ? Please give a explanation on this

Reply:
SCENARIO
APPLICATION PROFILE only for http,udp,tcp
PERSISTENCE PROFILE only for cookie,sourceip,generic
SSL PROFILE only for **client,server**
<img width="477" alt="EXAMPLE" src="https://user-images.githubusercontent.com/61727100/181068572-40296aed-1c6b-409d-ba65-461c122e06b2.PNG">
<img width="508" alt="ERR" src="https://user-images.githubusercontent.com/61727100/181068625-56033144-ce50-41f0-98d7-00e0265e86b0.PNG">